lib/fresh_connection/access_control.rb

```diff
@@ -62,6 +62,7 @@ def recovery?(spec_name)
       def catch_exceptions
         return @catch_exceptions if defined?(@catch_exceptions)
         @catch_exceptions = [ActiveRecord::StatementInvalid]
+        @catch_exceptions << ActiveRecord::ConnectionNotEstablished
```

Comment:
`ActiveRecord::ConnectionNotEstablished` has been defined since acrtiverecord 5.2

https://github.com/rails/rails/blob/8030cff808657faa44828de001cd3b80364597de/activerecord/lib/active_record/errors.rb#L46-L50